packages/pulp/python-pulpcore/python-pulpcore.spec

```diff
@@ -0,0 +1,94 @@
+# Created by pyp2rpm-3.3.3
+%global pypi_name pulpcore
+
+Name:           python-%{pypi_name}
+Version:        3.0.0rc8
+Release:        1%{?dist}
+Summary:        Pulp Django Application and Related Modules
+
+License:        GPLv2+
+URL:            http://www.pulpproject.org
+Source0:        https://files.pythonhosted.org/packages/source/p/%{pypi_name}/%{pypi_name}-%{version}.tar.gz
+BuildArch:      noarch
+
+BuildRequires:  python3-devel
+BuildRequires:  python3-setuptools
+
+%description
+Pulp is a platform for managing repositories of content, such as software
+packages, and pushing that content out to large numbers of consumers.
+
+Using Pulp you can:
+- Locally mirror all or part of a repository
+- Host your own content in a new repository
+- Manage content from multiple sources in one place
+- Promote content through different repos in an organized way
+
+%package -n     python3-%{pypi_name}
+Summary:        %{summary}
+%{?python_provide:%python_provide python3-%{pypi_name}}
+Requires:       python3-coreapi >= 2.3.3
+Conflicts:      python3-coreapi >= 2.4
+Requires:       python3-djangorestframework-queryfields >= 1.0.0
+Conflicts:      python3-djangorestframework-queryfields >= 1.1
```

Comment:
Is this Requires/Conflicts a newer convention? I am used to doing a greater and less than?

Reply:
Yes. See https://github.com/fedora-python/pyp2rpm/pull/196

Doing two requires as you say can result in different packages satisfying the requirement. On Fedora/EL8 we could use rich requires, but not here on EL7.

Reply:
I'm not getting it. Is there an example of how `Requires: python3-djangorestframework-queryfields < 1.1.0` results in a different dependency resolution than the conflicts? Consider this not a blocker, but more of I am interested to understand this RPM nuance. And also understand why we wouldn't adopt this pattern in other places.

Reply:
We ran into this when we were attempting to lock down python-celery and friends.  The short of it is, on yum based systems, nothing *just works* on enforcing the maximum versions, and using requires alone to enforce the issue causes misleading errors.  i.e. it will just say "nothing provides xxxx", while using an explicit conflicts will give you an error expressing the explicit conflicts.

It would seem that pyp2rpm's way of handling this issue is to forego the requires aspect of it for a conflicts only approach.

Reply:
My takeway is: both cases work properly with respect to yum, conflicts just gives an error output that is easier for a user to parse than double requires?

Reply:
"work properly" is arguable, they more "work identically".  With multiple repos, yum will never do the wrong thing, but it also will not always find the right one, necessitating either repo priority, or an alternative solution (hence the high epoch on the celery environment in pulp).

But yes, that would be a good assumption.